standalone/ft_benchmark/ft_benchmark.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * DRace, a dynamic data race detector
+ *
+ * Copyright 2020 Siemens AG
+ *
+ * Authors:
+ *   Mihai Robescu <mihai-gabriel.robescu@siemens.com>
+ *   Felix Moessbauer <felix.moessbauer@siemens.com>
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <algorithm>
+#include <atomic>
+#include <iomanip>
+#include <iostream>
+#include <mutex>
+#include <random>
+#include <set>
+#include <thread>
+#include <vector>
+
+
+std::mutex mx;
+static std::set<int> random_reads;
+static std::set<int> random_writes;
+static std::random_device rd{};
+std::mt19937 gen{0};
+
+void generate_block(int i,
+                    std::vector<std::pair<uintptr_t*, uintptr_t*>>* blocks) {
+  std::mt19937_64 gen(42 + i);
+  uintptr_t* ptr = new uintptr_t[i];
+
+  // fill block
+  std::generate(ptr, ptr + i, [&]() { return gen(); });
+  blocks->emplace_back(ptr, ptr + i);
+}
+
+void read_from_block(std::vector<std::pair<uintptr_t*, uintptr_t*>>* blocks) {
+  try {
+    std::uniform_int_distribution<int> dist(0, blocks->size() - 1);
+    int rnd = dist(gen);
+    random_reads.emplace(rnd);
+    uintptr_t* begin = (*blocks)[rnd].first;
+    uintptr_t* end = (*blocks)[rnd].second;
+
+    uintptr_t* iter = begin;
+    while (iter != end) {
+      auto tmp = *iter;
+      iter++;
+    }
+    // std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
If this was only used for debugging, try removing the comment if it is not relevant anymore.
Similarly in lines 74, 125, 134.